Automattic/woocommerce-payments
# Prevent WooPay redirection when OTP frame is closed

Fixes #6153

#### Changes proposed in this Pull Request

When the shopper is logged in to WooPay, and third-party cookies are enabled, opening WooPay OTP iframe will automatically initiate the checkout session and redirect them. Closing the iframe does not interrupt the process, so the shopper gets redirected even after closing the modal.
This PR aims to prevent that. I couldn't find a good way to abort the Promise, so I added a check for the modal to their `then`. It checks if there is any `.platform-checkout-otp-iframe` element in the document and interrupt the logic if there isn't. This will not interrupt the session initialization, but it will prevent the result after that, and I don't think we will have any problems with that.

#### Testing instructions

- Make sure you are logged in to WooPay.
- Ensure third-party cookies are enabled in your browser.
- Add an item to the cart and click the WooPay express checkout button. Wait until the modal opens with the "Taking you to checkout" spinner and close the modal by hitting esc.
    - When using `develop`, you will still get redirected to WooPay.
    - When using `prevent-redirect-when-closed`, you should not get redirected.

-------------------


**Post merge**
